frontend/src/lib/auth/permissions.test.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  STATUSES_ELIGIBLE_FOR_EDIT,
+  STATUSES_ELIGIBLE_FOR_REVIEW,
+} from "../../pages/destructionlist/detail/constants";
+import { User } from "../api/auth";
+import {
+  DESTRUCTION_LIST_STATUSES,
+  DestructionList,
+} from "../api/destructionLists";
+import {
+  canMarkListAsFinal,
+  canReviewDestructionList,
+  canStartDestructionList,
+  canUpdateDestructionList,
+  canViewDestructionList,
+} from "./permissions";
+
+/**
+ * Helper function to generate user objects with different permissions
+ * @param {boolean} canStartDestruction - Permission to start destruction
+ * @param {boolean} canReviewDestruction - Permission to review destruction
+ * @returns {User} - User object with specified permissions
+ */
+const generateUserWithPermissions = (
+  canStartDestruction: boolean,
+  canReviewDestruction: boolean,
+): User => ({
+  pk: 1,
+  username: "testuser",
+  firstName: "Test",
+  lastName: "User",
+  email: "testuser@example.com",
+  role: {
+    name: "Test Role",
+    canStartDestruction,
+    canReviewDestruction,
+    canViewCaseDetails: true,
+  },
+});
+
+describe("Permissions tests with all combinations", () => {
```

Comment:
Describe is the top level, I typically like to be the name of the test subject ("canStartDestructionList()")

Even though within describe there can be another describes, I suggest keeping a more flat structure:

current:
```
- describe("Some description")
  - describe("someFunction")
    - beforeEach()
    - afterEach()
    - test("some test case) 
    - test("some other case)
    
  - describe("someOtherFunction")
    - beforeEach()
    - afterEach()
    - test("some test case) 
    - test("some other case)
  
```

**suggested**:
```
├── describe("someFunction()")/
│   ├── beforeEach()
│   ├── afterEach()
│   ├── test("some test case) 
│   └── test("some other case)
└── describe("someOtherFunction()")/
    ├── beforeEach()
    ├── afterEach()
    ├── test("some test case) 
    └── test("some other case)
 ```